### PR TITLE
Update to make SSO JSConnect V3 compatible.

### PR DESCRIPTION
**Pull request by nervoustwit**
This PR will close this issue: https://github.com/vanilla/addons/issues/792

Recent changes to browser cookie policies have pushed us to rewrite the JSConnect plugin and all the libraries that handle authentication for JSConnect so that we can offer a version that does not rely on reading cookies.

This PR takes the basic functionality from PHPJSConnect library and retro-fits it to the Wordpress Vanilla plugin.

### CODE REVIEW

**Comment by nervoustwit** [comment]
### Testing

You will want to spin up a local version of Wordpress to test this. Wordpress requires an older version of PHP and MySql than what we run in Vanilla's docker environment. To create a container for Wordpress locally and test the Wordpress-Vanilla plugin follow these instructions:

 1. Create folder called `wordpress-local`.
 2. Clone the `wordpress-vanilla` repo into it.
 3. Create a `docker-compose.yml` file in `wordpress-local`
 4. Copy the yml content from below into it.
 5. Verify that the Volumes entry reflects the source path to your `wordpress-vanilla` plugin. 
```     
volumes: 
       - "/private/var/www/vanilla-workspace/wordpress-local/wordpress-vanilla:/var/www/html/wp-content/plugins"

```
 6. In docker desktop 
Screen Shot 2020-04-13 at 4 10 06 PM click on 'Preferences > Resources > File Sharing.
 7. Add the path to the `wordpress-vanilla` plugin.
 6. In a console, cd into  `/wordrpress-local/`
 7. type `$ docker-compose up -d`.
 8. In `vanilla-docker` do `$ docker-compose down` and then `$ docker-compose up --build` (so that Wordpress container can see the Vanilla network).
 9. Navigate to `http://localhost:8000` and configure Wordpress.
 10. Configure the plugin and point your dev.vanilla.localhost to Wordpress.

File: functions.php [hunk 1]
```diff
@@ -339,9 +339,12 @@ function vf_get_link_url($options) {
...
    global $current_user, $wp_roles;
 
-   if (!function_exists('get_currentuserinfo'))
+   if (!function_exists('get_currentuserinfo')) {
       require_once ABSPATH . WPINC . '/pluggable.php';
-   get_currentuserinfo();
+      get_currentuserinfo();
+   } else {
+       wp_get_current_user();
+   }
```
Comment (tburry) [inline functions.php:349]: This looks off.

File: sso.php [hunk 2]
```diff
@@ -1,4 +1,10 @@
 <?php
+ini_set('display_errors', 1);
+ini_set('display_startup_errors', 1);
+error_reporting(E_ALL|E_STRICT);
```
Comment (tburry) [inline sso.php]: This looks like debugging code.

File: sso.php [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
+
+require __DIR__.'/dist/functions.jsconnect.php';
```
Comment (tburry) [inline sso.php]: require_once

**Review by tburry — CHANGES_REQUESTED** [review]

File: functions.php (hunk 1 above)
Comment (nervoustwit) [reply functions.php:349]: get_currentuserinfo() throws a warning in wordpress. For backwards compatibility I left in because it still works.

**Comment by nervoustwit** [comment]
```
version: '3.5'

networks:
  vanilla_network:
    external: true

services:
   db:
     image: mysql:5.7
     volumes:
       - db_data:/var/lib/mysql
     restart: always
     environment:
       MYSQL_ROOT_PASSWORD: somewordpress
       MYSQL_DATABASE: wordpress
       MYSQL_USER: wordpress
       MYSQL_PASSWORD: wordpress
     networks: 
       - "vanilla_network"

   wordpress:
     depends_on:
       - db
     image: wordpress:latest
     ports:
       - "8000:80"
     restart: always
     environment:
       WORDPRESS_DB_HOST: db:3306
       WORDPRESS_DB_USER: wordpress
       WORDPRESS_DB_PASSWORD: wordpress
       WORDPRESS_DB_NAME: wordpress
     volumes: 
       - "/private/var/www/vanilla-workspace/wordrpress-local/wordpress-vanilla:/var/www/html/wp-content/plugins"
     networks: 
       - "vanilla_network"
volumes:
    db_data: {}
networks:
    vanilla_network:
        driver: "bridge"
        name: "vanilla_network"

```

**Review by tburry — APPROVED** [review]

**Review by initvector — APPROVED** [review]

**Pull request merged**